src/lib/dai.c

```diff
@@ -212,8 +224,47 @@ struct dai *dai_get(uint32_t type, uint32_t index, uint32_t flags)
 {
 	const struct device *dev;
 	struct dai *d;
+	uint32_t z_type;
 
-	dev = dai_get_zephyr_device(type, index);
+	switch (type) {
+	case SOF_DAI_INTEL_SSP:
+		z_type = DAI_INTEL_SSP;
+		break;
+	case SOF_DAI_INTEL_DMIC:
+		z_type = DAI_INTEL_DMIC;
+		break;
+	case SOF_DAI_INTEL_HDA:
+		z_type = DAI_INTEL_HDA;
+		break;
+	case SOF_DAI_INTEL_ALH:
+		z_type = DAI_INTEL_ALH;
+		break;
+	case SOF_DAI_IMX_SAI:
+		z_type = DAI_IMX_SAI;
+		break;
+	case SOF_DAI_IMX_ESAI:
+		z_type = DAI_IMX_ESAI;
+		break;
+	case SOF_DAI_AMD_BT:
+		z_type = DAI_AMD_BT;
+		break;
+	case SOF_DAI_AMD_SP:
+		z_type = DAI_AMD_SP;
+		break;
+	case SOF_DAI_AMD_DMIC:
+		z_type = DAI_AMD_DMIC;
+		break;
+	case SOF_DAI_MEDIATEK_AFE:
+		z_type = DAI_MEDIATEK_AFE;
+		break;
+	case SOF_DAI_INTEL_UAOL:
+		z_type = DAI_INTEL_UAOL;
+		break;
+	default:
+		return NULL;
+	}
```

Comment:
this looks like it should be a separate commit

Reply:
Moved this part to a separate commit. Thanks !